execution_engine/src/storage/trie_store/operations/mod.rs

```diff
@@ -505,41 +510,162 @@ where
     }
 }
 
+struct TrieScanRaw<K, V> {
+    tip: Either<Bytes, Trie<K, V>>,
+    parents: Parents<K, V>,
+}
+
+impl<K, V> TrieScanRaw<K, V> {
+    fn new(tip: Either<Bytes, Trie<K, V>>, parents: Parents<K, V>) -> Self {
+        TrieScanRaw { tip, parents }
+    }
+}
+
+/// Just like scan, however we don't parse the tip.
+fn scan_raw<K, V, T, S, E>(
+    _correlation_id: CorrelationId,
+    txn: &T,
+    store: &S,
+    key_bytes: &[u8],
+    root_bytes: Bytes,
+) -> Result<TrieScanRaw<K, V>, E>
+where
+    K: ToBytes + FromBytes + Clone,
+    V: ToBytes + FromBytes + Clone,
+    T: Readable<Handle = S::Handle>,
+    S: TrieStore<K, V>,
+    S::Error: From<T::Error>,
+    E: From<S::Error> + From<bytesrepr::Error>,
+{
+    let path = key_bytes;
+
+    let mut current_trie;
+    let mut current = root_bytes; //Bytes::from(root_bytes.to_bytes()?);
+    let mut depth: usize = 0;
+    let mut acc: Parents<K, V> = Vec::new();
+
+    loop {
+        current_trie = if current.first() == Some(&0) {
```

Comment:
Does `0` at first position have a special meaning? If so, can we used named constant instead of magic number?

Reply:
Good point, I was thinking about it. I added a `TrieTag` enum to represent it, and used it across the trie operations